### PR TITLE
Don't generate serializers for foreign types in Orleans.Streaming.EventHubs

### PR DESCRIPTION
**Pull request by ReubenBond**

### CODE REVIEW

**Comment by ReubenBond** [comment]
@dotnet-bot test functional

**Comment by ReubenBond** [comment]
@dotnet-bot test functional

**Comment by ReubenBond** [comment]
@dotnet-bot test functional

File: src/Azure/Orleans.Streaming.EventHubs/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -1,7 +1,3 @@
 ﻿using System.Runtime.CompilerServices;
-using Orleans.CodeGeneration;
-using Orleans.ServiceBus.Providers;
 
 [assembly: InternalsVisibleTo("ServiceBus.Tests")]
-
-[assembly: KnownAssembly(typeof(EventHubSequenceTokenV2), TreatTypesAsSerializable = true)]
```
Comment (jason-bragg) [inline src/Azure/Orleans.Streaming.EventHubs/Properties/AssemblyInfo.cs]: How is a serializer generated for EventHubSequenceTokenV2?  It's not used in any grain interfaces.  That's why it was previously marked with the GenerateSerializer attribute

Comment (ReubenBond) [reply src/Azure/Orleans.Streaming.EventHubs/Properties/AssemblyInfo.cs]: It's marked as `[Serializable]`

**Comment by ReubenBond** [comment]
@jason-bragg I added `GenerateSerializer` for `EventHubSequenceTokenV2` as an extra safety net: the build will fail if no serializer is generated. In addition, we also check that a test to drive confidence: `InternalSerializableTypesHaveSerializers` checks that a serializer exists at runtime for `EventHubSequenceTokenV2`

**Comment by jason-bragg** [comment]
@dotnet-bot test functional please

**Pull request merged**